### PR TITLE
Extensions/jdbc/mariadb: fix new Options class path registered for reflection

### LINKED ISSUE #6501: Data source Connection Issue After Updating Quarkus (0.25.0 to 1.1.1.Final)
**Describe the bug**
After updating Quarkus to 1.1.1.Final the bootstrap process fail with java.sql.SQLException saing Access denied. But it works without any issue with Quarkus 0.25.0 without any configuration change. Apparently quarkus doesn't detect the environment variables.  

**Expected behavior**
It should create the data-source connection successfully.

**Actual behavior**
I am using the quarkus native image in a docker container which connects to a mariadb database.
But when initializing it fails with
` java.sql.SQLException: Access denied for user 'root'@'172.17.42.1' (using password: NO)
`
in Mariadb log I can see an authentication error
`2020-01-10 13:17:28 69 [Warning] Access denied for user 'root'@'172.17.42.1' (using password: NO)`

Stacktrace
```
2020-01-10 18:47:27,158 host NativeImageGeneratorRunner[6486] INFO  [io.quarkus] (main) myapp stopped in 0.014s
2020-01-10 18:47:28,548 host NativeImageGeneratorRunner[32069] INFO  [org.fly.cor.int.lic.VersionPrinter] (main) Flyway Community Edition 6.1.0 by Redgate
2020-01-10 18:47:28,549 host NativeImageGeneratorRunner[32069] WARN  [io.agr.pool] (Agroal_16893719241) Datasource '<default>': Could not connect to address=(host=service1)(port=13306)(type=master) : Access denied for user 'root'@'172.17.42.1' (using password: NO)
2020-01-10 18:47:28,550 host NativeImageGeneratorRunner[32069] WARN  [io.agr.pool] (Agroal_16893719241) Datasource '<default>': Could not connect to address=(host=service1)(port=13306)(type=master) : Access denied for user 'root'@'172.17.42.1' (using password: NO)
2020-01-10 18:47:28,550 host NativeImageGeneratorRunner[32069] ERROR [io.qua.application] (main) Failed to start application: org.flywaydb.core.internal.exception.FlywaySqlException: 
Unable to obtain connection from database: Could not connect to address=(host=service1)(port=13306)(type=master) : Access denied for user 'root'@'172.17.42.1' (using password: NO)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 28000
Error Code : 1045
Message    : Could not connect to address=(host=service1)(port=13306)(type=master) : Access denied for user 'root'@'172.17.42.1' (using password: NO)

	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:60)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:80)
	at org.flywaydb.core.Flyway.execute(Flyway.java:437)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:147)
	at io.quarkus.flyway.runtime.FlywayRecorder.migrate(FlywayRecorder.java:54)
	at io.quarkus.flyway.runtime.FlywayRecorder.doStartActions(FlywayRecorder.java:35)
	at io.quarkus.deployment.steps.FlywayProcessor$configureRuntimeProperties61.deploy_0(FlywayProcessor$configureRuntimeProperties61.zig:79)
	at io.quarkus.deployment.steps.FlywayProcessor$configureRuntimeProperties61.deploy(FlywayProcessor$configureRuntimeProperties61.zig:98)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:234)
	at io.quarkus.runtime.Application.start(Application.java:87)
	at io.quarkus.runtime.Application.run(Application.java:210)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.sql.SQLInvalidAuthorizationSpecException: Could not connect to address=(host=service1)(port=13306)(type=master) : Access denied for user 'root'@'172.17.42.1' (using password: NO)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:239)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.connectWithoutProxy(AbstractConnectProtocol.java:1241)
	at org.mariadb.jdbc.internal.util.Utils.retrieveProxy(Utils.java:610)
	at org.mariadb.jdbc.MariaDbConnection.newConnection(MariaDbConnection.java:142)
	at org.mariadb.jdbc.Driver.connect(Driver.java:86)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:200)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:390)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:372)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:65)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1146)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:460)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.sql.SQLException: Access denied for user 'root'@'172.17.42.1' (using password: NO)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.authenticationHandler(AbstractConnectProtocol.java:729)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.createConnection(AbstractConnectProtocol.java:507)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.connectWithoutProxy(AbstractConnectProtocol.java:1236)
	... 13 more

```

This only happens with Quarkys 1.1.1.Final. I have configured Mariadb to allow remote connect.

**To Reproduce**

1. Create a Quarkus application which uses a datasource.
2. Provide the given environment variables when creating the docker container from the generated native image.

**Configuration**
```properties
quarkus.datasource.url = jdbc:mariadb://service1:13306
quarkus.datasource.driver = org.mariadb.jdbc.Driver
quarkus.datasource.username = root
quarkus.datasource.password = mypass
quarkus.hibernate-orm.database.default-schema=timepickle
```

```Bash
#!/bin/bash

docker run --name container1 \
    --add-host=service1:172.17.42.1 \
    -e QUARKUS_PROFILE=prod \
    -e QUARKUS_DATASOURCE_PASSWORD='mypass' \
    -e QUARKUS_DATASOURCE_URL='jdbc:mariadb://service1:13306' -itd -p 18080:8080 myapp:1.1

```
**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 4.15.0-1065-oem
 - Output of `java -version`: java version "1.8.0_191"
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)

 - GraalVM version (if different from Java): openjdk version "1.8.0_232"
OpenJDK Runtime Environment (build 1.8.0_232-20191008104205.buildslave.jdk8u-src-tar--b07)
OpenJDK 64-Bit GraalVM CE 19.2.1 (build 25.232-b07-jvmci-19.2-b03, mixed mode)

 - Quarkus version or git rev: 1.1.1.Final

### PR DESCRIPTION
**Pull request by pplr**
org.mariadb.jdbc.internal.util.Options changed to org.mariadb.jdbc.util.Options in mariadb-java-client version 2.5.2

Fix #6501

### CODE REVIEW

**Comment by geoand** [comment]
Thanks for the contribution @pplr !

Looks good to me but I think @Sanne should be the one to have the final say

**Comment by gsmet** [comment]
We usually try to avoid loading the classes but I think the ones from the MariaDB and MySQL drivers are moving a lot as it's not the first breakage we have and it's probably safer to load them.

@Sanne WDYT?

**Comment by Sanne** [comment]
> We usually try to avoid loading the classes but I think the ones from the MariaDB and MySQL drivers are moving a lot as it's not the first breakage we have and it's probably safer to load them.

I would prefer we stick to a single principle. Didn't we have any test failure catching this? Improving test coverage is probably the better way forward.

**Comment by pplr** [comment]
jpa-mariadb integration tests in native mode are failing but it seems not to be executed by the pipeline?

**Comment by pplr** [comment]
May I add a "start mariadb" step in ci-templates/native-build-steps.yaml  and a `-Dtest-mariadb`?

**Comment by Sanne** [comment]
@pplr it used to be there already. I will need to ask around to figure out why it was disabled: we had problems with CI capacity, I guess it was done because of that.

I think we won't be able to run these tests all the time, but we should have it at least periodically (nithgly?)

**Review by gastaldi — APPROVED** [review]

**Comment by pplr** [comment]
@Sanne do you think that it is relevant to avoid loading the class and add unit testing for `MariaDBJDBCReflections` class?

I tried to write some, but it seems a bit weird.

```
package io.quarkus.jdbc.mariadb.deployment;

import static org.junit.jupiter.api.Assertions.*;

import io.quarkus.deployment.annotations.BuildProducer;
import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;

class MariaDBJDBCReflectionsTest {

    private static class BuildProducerReflectiveVerifier implements BuildProducer<ReflectiveClassBuildItem> {
        @Override
        public void produce(ReflectiveClassBuildItem item) {
            for (String className : item.getClassNames()) {
                assertClassExists(className);
            }
        }
    }

    private static class BuildProducerRuntimeInitializedVerifier implements BuildProducer<RuntimeInitializedClassBuildItem>{
        @Override
        public void produce(RuntimeInitializedClassBuildItem item) {
            assertClassExists(item.getClassName());
        }
    }

    public static void assertClassExists(String className){
        ClassLoader classLoader = MariaDBJDBCReflectionsTest.class.getClassLoader();
        assertDoesNotThrow(() -> classLoader.loadClass(className));
    }

    @org.junit.jupiter.api.Test
    void build() {
        MariaDBJDBCReflections r = new MariaDBJDBCReflections();
        BuildProducerReflectiveVerifier p = new BuildProducerReflectiveVerifier();
        r.build(p);
    }

    @org.junit.jupiter.api.Test
    void runtimeInit(){
        MariaDBJDBCReflections r = new MariaDBJDBCReflections();
        BuildProducerRuntimeInitializedVerifier p = new BuildProducerRuntimeInitializedVerifier();
        r.runtimeInit(p);
    }
}
```

**Comment by gsmet** [comment]
If the native test was failing then it's fine by me and we don't need more tests.

It was mistakenly not run in CI, I'm working on it.

Could you go back to not loading the classes then? Thanks!

**Comment by Sanne** [comment]
> If the native test was failing then it's fine by me and we don't need more tests.

+1

> It was mistakenly not run in CI, I'm working on it.

Thanks @gsmet ! I was confused about that, glad you figured out it was a mistake. Will we get back to all checks on each PR or rather have it build master periodically?

> Could you go back to not loading the classes then? Thanks!

+1 many thanks @pplr

**Comment by gsmet** [comment]
@pplr can you squash everything into a single commit, please?

Thanks!

**Comment by gsmet** [comment]
I squashed the 2 commits and rebased on master as there was an unrelated CI issue.

Let's wait for CI.

**Review by gsmet — APPROVED** [review]

**Pull request merged**
